src/utils.js

```diff
@@ -239,3 +241,60 @@ export const validateDropDownValue = (shouldValidate) => (value) => {
 
   return undefined;
 };
+
+export const getUserPreferences = (mediatedRequest, userData, servicePoints) => {
+  const userPreferences = {};
+
+  if (mediatedRequest) {
+    if (isDeliverySelected(mediatedRequest.fulfillmentPreference)) {
+      const address = userData?.users[0].personal.addresses.find(({ addressTypeId }) => addressTypeId === mediatedRequest.deliveryAddressTypeId);
+
+      userPreferences.label = <FormattedMessage id="ui-requests-mediated.requesterDetails.deliveryAddress" />;
+      userPreferences.value = <AddressDetails address={address} />;
+    } else {
+      userPreferences.label = <FormattedMessage id="ui-requests-mediated.requesterDetails.pickupServicePoint" />;
+      userPreferences.value = servicePoints?.servicepoints.find(servicePoint => servicePoint.id === mediatedRequest.pickupServicePointId)?.name;
+    }
+  }
+
+  return userPreferences;
+};
+
+export const getReferredRecordData = (mediatedRequest) => {
+  if (mediatedRequest) {
```

Comment:
Again, small notes: For better readability, it is better to keep the main login not wrapped in "if" stat and jest use guards with early returns like:
<img width="719" alt="Screenshot 2024-08-07 at 17 57 29" src="https://github.com/user-attachments/assets/fc8bb512-ab3f-44d0-9d79-481457ffcb5b">